sdk/kiosk/src/client/tp-manager.ts

```diff
@@ -0,0 +1,371 @@
+	 * This can be called after calling the `create` function to share the `TransferPolicy`,
+	 * and transfer the `TransferPolicyCap` to the specified address
+	 */
+	shareAndTransferCap(tx: TransactionBlock, address: string) {
+		if (!this.itemType || !this.policyCap || !this.policyId)
+			throw new Error('This function can only be called after `transferPolicyManager.create`');
+
+		shareTransferPolicy(tx, this.itemType, this.policyId as TransactionArgument);
+		tx.transferObjects([this.policyCap as TransactionArgument], tx.pure(address, 'address'));
+
+		this.#resetPolicy();
+	}
+
+	#resetPolicy() {
+		this.policyCap = undefined;
+		this.policyId = undefined;
+	}
+
+	/**
+	 * Find the Policy Cap Object for a specified address.
+	 * Returns null if the address doesn't own it.
+	 */
+	async getPolicyCapId(itemType: string, address: string) {
+		return queryOwnedTransferPolicyCap(this.client, address, itemType);
+	}
+
+	/**
+	 * Setup the TransferPolicy object by passing the itemType and the owner address.
+	 * @param itemType The Type for which we're managing the transfer policy
+	 * @param address The owner of the Cap.
+	 */
+	async setPolicyByTypeAsync(itemType: string, address: string) {
+		let cap = await this.getPolicyCapId(itemType, address);
+		if (!cap)
+			throw new Error(
+				`Couldn't find a TransferPolicyCap for type ${itemType} owned by address ${address}`,
+			);
+
+		return this.setPolicyAsync(cap);
+	}
+
+	/**
+	 * Setup the TransferPolicy object by passing just the policyCapId.
+	 * It automatically finds the policyId, as well as it's type.
+	 * @param policyCapId The Object ID for the TransferPolicyCap object
+	 */
+	async setPolicyAsync(policyCapId: string) {
+		let capObject = await this.client.getObject({
+			id: policyCapId,
+			options: {
+				showContent: true,
+			},
+		});
+		if (!capObject) throw new Error("This cap Object wasn't found");
+
+		let type = (capObject?.data?.content as { type: string })?.type;
+		//@ts-ignore-next-line
+		let policyId = capObject?.data?.content?.fields?.policy_id;
+
+		if (!type.includes(TRANSFER_POLICY_CAP_TYPE))
+			throw new Error('Invalid Cap Object Id. Are you sure this ID is a cap?');
+
+		// Transform 0x2::transfer_policy::TransferPolicyCap<itemType> -> itemType
+		let itemType = type.replace(TRANSFER_POLICY_CAP_TYPE + '<', '').slice(0, -1);
+
+		this.setPolicy(policyId, policyCapId, itemType);
+	}
+
+	/**
+	 * Set Policy by passing the types / ids manually.
+	 * Use `setPolicyAsync` to automatically fetch them by just passing the Cap's object Id.
+	 * @param policyId The `TransferPolicy` Object ID (or object ref)
+	 * @param policyCap The `TransferPolicyCap` Object ID (or object ref)
+	 * @param type The `T` (type) for the `TransferPolicy`
+	 */
+	setPolicy(policyId: ObjectArgument, policyCap: ObjectArgument, type: string) {
+		this.setPolicyId(policyId);
+		this.setPolicyCap(policyCap);
+		this.setPolicyType(type);
+	}
+
+	setPolicyId(id: ObjectArgument) {
+		this.policyId = id;
```

Comment:
would probably be convenient if these methods returned `this` for chaining the same way the `add` methods do for consistency

Reply:
True, thanks!